migrations/20200403112052_map_prototcol_fate.js

```diff
@@ -0,0 +1,67 @@
+const transform = version => {
+  if (!version) {
+    return;
+  }
+  if (!version.protocols || !Array.isArray(version.protocols)) {
+    return version;
+  }
+  version.protocols.forEach(protocol => {
+    if (!protocol || !protocol['killing-method'] || !Array.isArray(protocol['killing-method'])) {
+      return;
+    }
+    if (protocol['killing-method'].includes('other')) {
+      return protocol['non-schedule-1'] = true;
+    }
+    if (protocol['killing-method'].includes('schedule-1')) {
+      return protocol['non-schedule-1'] = false;
+    }
+
+  });
+  return version;
+};
+
+exports.transform = transform;
+
+exports.up = function(knex) {
+  return Promise.resolve()
+    .then(() => {
+      return knex
+        .select('id')
+        .from('projects')
+        .where({ 'schema_version': 1 })
```

Comment:
As on #291 please fetch the versions as the top level query to avoid holding multiple versions in memory, and add some logging.

Thanks.